pkg/sql/fingerprint_span.go

```diff
@@ -145,15 +156,18 @@ func (p *planner) fingerprintSpanFanout(
 		ssts        [][]byte
 		fingerprint uint64
 	}{
-		ssts: make([][]byte, 0, len(workerPartitions)),
+		ssts: make([][]byte, 0, len(spanPartitions)),
 	}
 
 	grp := ctxgroup.WithContext(ctx)
-	for i := range workerPartitions {
-		workerIdx := i
+	for range maxWorkerCount {
 		grp.GoCtx(func(ctx context.Context) error {
-			spans := workerPartitions[workerIdx].Spans
-			for _, sp := range spans {
+			// Run until channel is empty
+			for {
+				sp, ok := <-spanChannel
```

Comment:
Probably want to `select` on the spanCh and on ctx.Done() in case we cancel while there is still work in the ch